src/libraries/Common/tests/Tests/System/StringTests.cs

```diff
@@ -2616,7 +2618,7 @@ public static void Format_Invalid()
 #pragma warning restore IDE0043 // Format string contains invalid placeholder
         }
 
-        [Theory]
+        [ConditionalTheory]
```

Comment:
Yes, this was exactly to be able to skip the test at runtime based on the input. Basically by wrapping it under a `ConditionalTheory` with no arguments it will not skip the test but will wrap the testcase with this class: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.XUnitExtensions/src/SkippedTheoryTestCase.cs#L23 which sets the message bus to be an instance of `SkippedMessageBus` which checks if the exception thrown is a `SkipTestException`, https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.XUnitExtensions/src/SkippedTestMessageBus.cs#L38 -- if that is the case then that test case is treated as skipped instead of failing and will print the exception message as the skipped reason to the console.

Reply:
Btw, it doesn't have to be a parameterless `ConditionalTheory`, but since in this case we want the test case to always run and just skip one inline data case, I did it that way.

Reply:
Thanks for the context!